server/db.ts

```diff
@@ -1,7 +1,7 @@
 import { MongoClient } from "mongodb";
 import type { NextApiRequest, NextApiResponse } from "next";
 
-export type AnimalDoc = {
+export type Animal = {
   name: string;
   id: string;
   ImgSrc: string;
```

Comment:
Maybe instead of building a new type, you could use `<Pick>` to build a new type based on the type `AnimalCardProps`
Link: https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys

If you want to do this, feel free to ask me :)